File: Assignment2_Section2/memory_management.cpp
#include <iostream>

void printArray(int* arr, int size) {
    for (int i = 0; i < size; ++i)
        std::cout << arr[i] << " ";
    std::cout << std::endl;
}

int main() {
    int* data = new int[5]; 
    for (int i = 0; i < 5; ++i)
        data[i] = i + 1;

    printArray(data, 5);

    delete[] data; 

    return 0;
}

File: Assignment2_Section2/MemoryManagement.java
public class MemoryManagement {
    public static void main(String[] args) {
        int[] data = new int[5]; 
        for (int i = 0; i < data.length; i++) {
            data[i] = i + 1;
        }

        printArray(data);
        data = null; 
        System.gc(); 
    }

    public static void printArray(int[] arr) {
        for (int value : arr) {
            System.out.println(value);
        }
    }
}
